packages/ui/components/scheduler/components/Scheduler.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useEffect, useMemo, useRef } from "react";
+
+import { useSchedulerStore } from "../state/store";
+import "../styles/styles.css";
+import { SchedulerComponentProps } from "../types/state";
+import { getDaysBetweenDates, getHoursToDisplay } from "../utils";
+import { BlockedList } from "./blocking/BlockedList";
+import { DateValues } from "./dateValues/DateValues";
+import { EmptyCell } from "./event/Empty";
+import { EventList } from "./event/EventList";
+import { SchedulerColumns } from "./grid";
+import { SchedulerHeading } from "./heading/SchedulerHeading";
+import { HorizontalLines } from "./horizontalLines";
+import { VeritcalLines } from "./verticalLines";
+
+export default function Scheduler(props: SchedulerComponentProps) {
+  const container = useRef<HTMLDivElement | null>(null);
+  const containerNav = useRef<HTMLDivElement | null>(null);
+  const containerOffset = useRef<HTMLDivElement | null>(null);
+  const initalState = useSchedulerStore((state) => state.initState);
+
+  const { startDate, endDate, startHour, endHour, usersCellsStopsPerHour } = useSchedulerStore((state) => ({
+    startDate: state.startDate,
+    endDate: state.endDate,
+    startHour: state.startHour || 0,
+    endHour: state.endHour || 23,
+    usersCellsStopsPerHour: state.gridCellsPerHour || 4,
+  }));
+
+  const days = useMemo(() => getDaysBetweenDates(startDate, endDate), [startDate, endDate]);
+
+  const hours = useMemo(() => getHoursToDisplay(startHour || 0, endHour || 23), [startHour, endHour]);
+
+  const numberOfGridStopsPerDay = hours.length * usersCellsStopsPerHour;
+
+  // Initalise State on inital mount
+  useEffect(() => {
+    initalState(props);
+  }, [props, initalState]);
+
+  return (
+    <MobileNotSupported>
+      <div
+        className="scheduler-wrapper flex h-full w-full flex-col overflow-y-scroll"
+        style={
+          { "--one-minute-height": `calc(1.75rem/(60/${usersCellsStopsPerHour}))` } as React.CSSProperties // This can't live in the css file because it's a dynamic value and css variable gets super
```

Comment:
Define a global CSS var we can use across the app. This has to live here because css vars don't really support dynamic variables and makes life a bit awkward